### ISSUE TITLE
Rendering SVG's from scripted backgroundImage-Property

### ISSUE THREAD

**Issue by more-vikr** [eclipse/birt#865]
How is it possible to output an SVG file rendered into PDF when the backgroundImage-Property is assigned in OnCreate-Script?

The Image-Item works with embedded and external resources, but this is not a good option for us, because we have to change many big reports.

### Steps to reproduce:

Using runtime org.eclipse.birt.runtime_4.9.0-20220308

• Create an empty report with a grid item.

_• Optional: Add a report variable "X" with contains the `<PATH TO YOUR SVG>`_

• In the OnCreate-Script of the Grid-Cell put following line:

`this.getStyle().backgroundImage="<PATH-TO-YOUR-SVG>";`

Replace `<PATH-TO-YOUR-SVG>` with your own SVG file.

_Optional: `this.getStyle().backgroundImage=vars["X"];`_

• Run the report

### Comment:

In the old runtime org.eclipse.birt.runtime_4.3.2.v20140225-1404 which runs on JRE8 it all worked fine.
I read that from Java11 and upwards SVG rendering is not 100% compatible with the actual BIRT runtime.
Also there may be a relation with #805 ?

**Comment by claesrosell** [comment]
This may be caused by a missing Batik bundle. I will try to look into it tonight.

**Comment by claesrosell** [comment]
Yes, org.apache.batik.codec is missing from the runtime build. Adding it to the .\birt-runtime\ReportEngine\lib\ folder solves the problem. I will try to figure out how the runtime package is built.

**Comment by claesrosell** [comment]
@more-vikr, can you test the builds from #866 to see if this fix works? Available here : https://github.com/eclipse/birt/actions/runs/1989569533

**Comment by claesrosell** [comment]
BTW, this bug will probably affect any rendering of SVG -> PNG in when using the different runtimes, since the Batik bundle was missing in a lot of features and most specifically in the engine runtime, since different ant scripts were copying jars from the engine target/package to various other places.

**Comment by claesrosell** [comment]
@hvbtup : I thought I did. Is not the "POJO Runtime" the "birt-runtime" artifact? Or am I missing something. I am extremely new to these runtimes. This is actually the first time I have worked with them

**Comment by hvbtup** [comment]
You are right. I'm too busy to be really focused, it seems. Some text with OSGI catched my eye, so I thought you fixed only the OSGI runtime. But looking at your commit more closely, you added it to the POJO runtime as well (and the artifact contains the expected JAR file in ReportEngine/lib.

About the different runtimes: AFAIK the code base is exactly the same. The difference seems to be just a single environment variable (BIRT_HOME IIRC), which is used to decide in the code to switch to OSGI mode or not. And the layout of the libraries is different.

**Comment by more-vikr** [comment]
> @more-vikr, can you test the builds from #866 to see if this fix works? Available here : https://github.com/eclipse/birt/actions/runs/1989569533

Hi @claesrosell, I just testend the build. The rendering of svg background definition now works as expected.
